core/src/kura.rs

```diff
@@ -305,224 +514,9 @@ pub enum Error {
-                file_string
-            );
-        }
-        self.path.join(dir_name).join(file_name)
-    }
-
-    /// Returns a stream of deserialized blocks that in order of reading from sorted storage files
-    ///
-    /// # Errors
-    /// * Will propagate errors from any FS operations
-    /// * Will provide error on non-sequential block (swapped, gap, not 1-based sequence)
-    ///
-    pub async fn read_all(
-        &self,
-    ) -> Result<impl Stream<Item = Result<VersionedCommittedBlock>> + '_> {
-        let io = self.io.clone();
-        let base_indices = storage_files_base_indices(&self.path, &self.io).await?;
-        let mut files = Vec::new();
-        for index in &base_indices {
-            files.push(self.filename_to_path(&index.to_string()));
-        }
-
-        let result = tokio_stream::iter(files)
-            .map(move |e| e)
-            .then(move |e| {
-                let io = io.clone();
-                // async move { io.open(e.into_os_string()).await }
-                async move { io.open(e.into()).await }
-            })
-            .map_ok(BufReader::new)
-            .map_ok(Self::read_file)
-            .try_flatten()
-            .enumerate()
-            .map(|(i, b)| b.map(|bb| (i, bb)))
-            .and_then(|(i, b)| async move {
-                if b.header().height == (i as u64) + 1 {
-                    Ok(b)
-                } else {
-                    Err(Error::InconsequentialBlockRead(b.header().height))
-                }
-            });
-        Ok(result)
-    }
-}
-
-/// Returns sorted Vec of datafiles base heights as u64
-///
-/// # Errors
-/// Will fail on filesystem access error
-async fn storage_files_base_indices<IO: DiskIO>(
-    path: &Path,
-    io: &IO,
-) -> Result<BTreeSet<NonZeroU64>> {
-    let dirs: Vec<(String, IO::Dir)> = io
-        .read_dir(path.to_path_buf())
-        .await?
-        .filter_map(|item| async {
-            if let Ok(item) = item {
-                let dir_name = item.to_string_lossy().into_owned();
-                match io.read_dir(path.join(dir_name.clone())).await {
-                    Ok(v) => Some((dir_name, v)),
-                    Err(_) => None,
-                }
-            } else {
-                None
-            }
-        })
-        .collect()
-        .await;
-
-    let mut bases = BTreeSet::<NonZeroU64>::new();
-    for (dir_string, dir) in dirs {
-        let files_vec: Vec<io::Result<OsString>> = dir.collect().await;
-        for file_string in files_vec.into_iter().flatten() {
-            let maybe_num =
-                format!("{}{}", dir_string, file_string.to_string_lossy()).parse::<NonZeroU64>();
-            if let Ok(num) = maybe_num {
-                bases.insert(num);
-            }
-        }
-    }
-
-    Ok(bases)
+    /// A read, which was initiated by the block store, went off the end of the block data file.
+    #[error("Block read out of bounds: {0}, {1}.")]
```

Comment:
maybe something like this?

```suggestion
    #[error("Tried reading block data out of bounds: {0}, {1}.")]
```